### ISSUE TITLE
High Contrast on Windows

### ISSUE THREAD

**Issue by default0** [buggins/dlangui#149]
I am using a high contrast design on windows. Not sure if the barely-readable text/background composition is DlangIDEs or DlangUIs fault, but I'd think it's an issue with DlangUI.
Screenshot:
![Image of the problem](http://puu.sh/mGzoy/1bf80d1d9b.png)

Not sure where to look or change something for a fix, but pretty sure the text is intended to be readable.

**Comment by buggins** [comment]
How do I reproduce it?

**Comment by default0** [comment]
Change the design you are using on Windows to Contrast Black from the list of Designs in the Control Panel.

**Comment by buggins** [comment]
I don't see Designs in Control Panel - on Win 7.
Looks like it's some new Win10 feature

**Comment by default0** [comment]
Nope, this issue was found with Win 7.
I'll give more detailed instructions on how to reproduce when I'm back home from work.

**Comment by default0** [comment]
Not yet home from work, but here's how you change:
Control Panel > Appearance and Personalization > Change the theme > Scroll down in list to the ones with high contrast > pick High Contrast Black

After that you should get visuals like the ones shown in the screenshot.

**Comment by buggins** [comment]
I've set High Contrast Black theme, but I don't see any difference in DlangUI app's content rendering. Image doesn't depend on desktop theme. I see only changes in window frame.
(checked on Win7)

**Comment by default0** [comment]
That seems odd - the specific screenshot I made was from a file open dialog in DlangIDE - so maybe this is specific to DlangIDE? I'm really confused by this.
I do know that many many GUI applications look weird with that theme and look fine without, though.
I'll try and look into this myself when I'm back home.

**Comment by buggins** [comment]
Btw, if you want high contrast black theme in DlangIDE, go to Edit/Preferences/UI - select Dark theme

**Comment by default0** [comment]
Oh my.
I did some digging around on this - having seen lots of GUI applications create similar visuals due to my selected theme I immediately accounted it to that - but apparently what is going on is that the Dark Theme in DlangIDE (which is what I have selected) changes the text Color to #E0E0E0 which is pretty close to white and obviously intended for a dark background. However, the background color of the widget used in file open dialogs stays unaffected by the theme - therefore remaining white, making the text barely readable.

So, in conclusion: It's an issue with DlangIDE's dark theme, not DlangUI's handling of anything!
Sorry for the false report, will go dig around the theme stuff in DlangIDE now and see if I can't fix it by making some adjustments to ide_theme_dark.xml.

**Comment by buggins** [comment]
Probably, it's DlangUI issue - improper handling of theme change in dialogs
